qiskit_experiments/library/characterization/analysis/drag_analysis.py

```diff
@@ -150,25 +150,25 @@ def _generate_fit_guesses(
         freq_bound = max(10 / user_opt.p0["freq0"], max(x_data))
 
         user_opt.bounds.set_if_empty(
-            amp=(-2 * max_abs_y, 0),
+            amp=(-2 * max_abs_y, 2 * max_abs_y),
             freq0=(0, np.inf),
             freq1=(0, np.inf),
             freq2=(0, np.inf),
             beta=(-freq_bound, freq_bound),
             base=(-max_abs_y, max_abs_y),
         )
-        user_opt.p0.set_if_empty(base=0.5)
+        user_opt.p0.set_if_empty(base=(user_opt.p0["amp"] or max_abs_y) / 2)
 
         # Drag curves can sometimes be very flat, i.e. averages of y-data
         # and min-max do not always make good initial guesses. We therefore add
         # 0.5 to the initial guesses. Note that we also set amp=-0.5 because the cosine function
         # becomes +1 at zero phase, i.e. optimal beta, in which y data should become zero
         # in discriminated measurement level.
         options = []
-        for amp_guess in (0.5, -0.5):
-            for beta_guess in np.linspace(min_beta, max_beta, 20):
+        for amp_factor in (-1, -0.5, 0.5, 1):
+            for beta_guess in np.linspace(0.5 * min_beta, 0.5 * max_beta, 10):
```

Comment:
I reverted to the full range. I was concerned this would take too much time. I solved the time issues by reducing the number of free frequency parameters in the fit from three to one.